pi.py

```diff
@@ -1 +1,578 @@
+#             If = 0 (BE02 default), NaN values in profile are ignored and PI is still calcuated
+#             If = 1, given NaN values PI will be set to missing (with IFLAG=3)
+#             NOTE: If any missing values are between the lowest valid level and ptop
+#             then PI will automatically be set to missing (with IFLAG=3)
+#
+#  OUTPUT:  VMAX is the maximum surface wind speed (m/s)
+#             reduced to reflect surface drag via V_reduc
+#
+#           PMIN is the minimum central pressure (hPa)
+#
+#           IFL is a flag: A value of 1 means OK; a value of 0
+#             indicates no convergence; a value of 2
+#             means that the CAPE routine failed to converge;
+#             a value of 3  means the CAPE routine failed due to
+#             missing data in the inputs
+#
+#           TO is the outflow temperature (K)
+#
+#           LNB is the level of neutral bouyancy where the outflow temperature
+#             is found (hPa), i.e. where buoyancy is actually equal to zero under the 
+#             condition of an air parcel that is saturated at sea level pressure
+#
+    
+    # convert units
+    SSTK=SSTC+273.15 # SST in kelvin
+    T=T+273.15       # Temperature profile in kelvin
+    R=R*0.001        # Mixing ratio profile in gm/gm
+
+    # CHECK 1: do SSTs exceed 5C? If not, set IFL=0 and return missing PI
+    if (SSTC <= 5.0):
+        VMAX=np.nan
+        PMIN=np.nan
+        IFL=0
+        TO=np.nan
+        LNB=np.nan
+        return(VMAX,PMIN,IFL,TO,LNB)
+
+    # CHECK 2: do Temperature profiles exceed 100K? If not, set IFL=0 and return missing PI
+    if (np.min(T) <= 100):
+        VMAX=np.nan
+        PMIN=np.nan
+        IFL=0
+        TO=np.nan
+        LNB=np.nan
+        return(VMAX,PMIN,IFL,TO,LNB)
+    
+    # Set Missing mixing ratios to zero gm/gm, following Kerry's algorithm
+    R[np.isnan(R)]=0.
+    
+    # Saturated water vapor pressure
+    # from Clausius-Clapeyron relation/August-Roche-Magnus formula
+    ES0=6.112*np.exp(17.67*SSTC/(243.5+SSTC))
+
+    # Constants
+    NK=0         # level from which parcels lifted (first pressure level)
+    b=2.0        # Exponent for estimating azimuthal velocity in the eye, V=V_m(r/r_m)**b (Emanuel 1995, EQN. 25)
+    ptop=50      # Pressure below which sounding is ignored (hPa)
+    RD=287.04    # [J/kg.K] gas constant of dry air
+    EPS=RD/461.5 # [unitless] epsilon, the ratio of gas constants
+    
+    #
+    #   ***   Find environmental CAPE *** 
+    #
+    TP=T[NK]
+    RP=R[NK]
+    PP=P[NK]
+    result = cape(TP,RP,PP,T,R,P,ascent_flag,ptop,miss_handle)
+    CAPEA = result[0]
+    IFLAG = result[3]
+    # if the CAPE function tripped a flag, set the output IFL to it
+    if (IFLAG != 1):
+        IFL=int(IFLAG)
+    
+    #
+    #   ***   Begin iteration to find mimimum pressure   ***
+    #
+    
+    # set loop counter and initial condition
+    NP=0         # loop counter
+    PM=970.0
```

Comment:
What is this? 970?

Reply:
It's an initial condition for the iteration that was chosen arbitrarily by Kerry.